packages/views/shared/discreetMode/discreetMode.js

```diff
@@ -1,46 +1,43 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import routes from '@screens/router/routes';
 import { selectSearchParamValue } from '@screens/router/searchParams';
 import styles from './discreetMode.css';
 
-class DiscreetMode extends Component {
-  handleBlurOnOtherWalletPage() {
-    const { account, location: { search }, token } = this.props;
+const DiscreetMode = ({
+  children, location, isDiscreetMode, shouldEvaluateForOtherAccounts,
+  addresses, account, token,
+}) => {
+  const handleBlurOnOtherWalletPage = () => {
+    const { search } = location;
     const address = selectSearchParamValue(search, routes.account.searchParam);
     return account.info && address === account.info[token].address;
-  }
+  };
```

Comment:
maybe we can change the function to the variable and memoized it also we can rename this function to the "isSomthing" because it just return a boolean

Reply:
@soroushm Resolved